lib/internal/fs/promises.js

```diff
@@ -252,6 +253,42 @@ class FileHandle extends EventEmitterMixin(JSTransferable) {
     return readable;
   }
 
+  /**
+   * @typedef {import('./streams').ReadStream
+   * } NodeJSReadStream
+   * @param {{
+   *   flags?: string;
+   *   encoding?: string;
+   *   autoClose?: boolean;
+   *   emitClose?: boolean;
+   *   start: number;
+   *   end?: number;
+   *   highWaterMark?: number;
+   *   fs?: Object | null;
+   *   }} [options]
+   * @returns {NodeJSReadStream}
+   */
+  readStream(options = undefined) {
+    return new ReadStream(undefined, { ...options, fd: this[kFd] });
```

Comment:
This is too simplistic as options/resource lifetime management. The `autoClose` option should be applied differently here, otherwise this is inherently broken because it would imply that two independent objects own the same fd. (The created stream needs to take ownership of the `FileHandle` object.)

The `fs` option does not make sense to pass at all, because the stream must use the FileHandle's read/write methods for I/O instead of using the plain fs read/write methods (which it currently does not).

Reply:
> This is too simplistic as options/resource lifetime management. The `autoClose` option should be applied differently here, otherwise this is inherently broken because it would imply that two independent objects own the same fd. (The created stream needs to take ownership of the `FileHandle` object.)

Do you have a test case where that problem comes up please?

> The `fs` option does not make sense to pass at all, because the stream must use the FileHandle's read/write methods for I/O instead of using the plain fs read/write methods (which it currently does not).

Agreed, I've fixed that by giving the `FileHandle` instance directly to the stream constructor.

Reply:
> > This is too simplistic as options/resource lifetime management. The `autoClose` option should be applied differently here, otherwise this is inherently broken because it would imply that two independent objects own the same fd. (The created stream needs to take ownership of the `FileHandle` object.)
> 
> Do you have a test case where that problem comes up please?

I was talking about these things: https://github.com/nodejs/node/blob/1c4df352b77514e93a46fb27e996755d45a0b530/lib/internal/fs/streams.js#L139-L141

However, I didn’t realize that `fs.createReadStream()`/`.createWriteStream()` already support `FileHandle`s and that the code for that was already there :)